### PR TITLE
Save/load sound and music settings from localStorage.

### PR DESCRIPTION
**Pull request by sysrage**

### CODE REVIEW

File: patcher/src/services/session/sounds.ts [hunk 1]
```diff
@@ -20,53 +20,49 @@ const UNMUTE_MUSIC = 'cse-patcher/sounds/UNMUTE_MUSIC';
...
-}
-
-const initialState = JSON.parse(localStorage.getItem(localStorageKey)) || false;
+const initialState = JSON.parse(localStorage.getItem(localStorageKey)) || {
+  playMusic: true,
+  playSound: true
+};
 
-export default function reducer(state: SoundsState = getInitialState(), action: SoundsAction = defaultAction): SoundsState {
+export default function reducer(state: SoundsState = initialState, action: SoundsAction = defaultAction): SoundsState {
```
Comment (codecorsair) [inline patcher/src/services/session/sounds.ts]: getInitialState should still be a function.  Also JSON.parse could through an exception if the data in local storage is not null but not valid, so need to safely try to parse that and catch any exceptions.

**Comment by sysrage** [comment]
Will do.

**Comment by sysrage** [comment]
PR updated to verify validity of saved settings.

**Comment by sysrage** [comment]
PR updated again based on feedback from JB.

**Comment by Mehuge** [comment]
Tested, works.  Looks good.

**Pull request merged**